CRM/Core/DAO.php

```diff
@@ -3130,4 +3141,14 @@ private function clearDbColumnValueCache() {
     }
   }
 
+  /**
+   * Return a mapping from field-name to the corresponding key (as used in fields()).
+   *
+   * @return array
+   *   Array(string $name => string $uniqueName).
+   */
+  public static function fieldKeys() {
```

Comment:
This is different to other functions....

```
  public static function &fieldKeys() {
```

I think the missing, hated, & will cause problems. Also if removing the caching matters at all it will matter a log as this is iterated through a lot....

Also - this function could maybe just have a different name & then the static here but not in line 2745 wouldn't matter? Or we should use self:: above

Reply:
1. In my testing, there was no issue having an `&` in the child class version of this function but not in the parent. PHP doesn't complain. But remember the goal is to get rid of the child class functions entirely, so even if future versions of PHP would start complaining about this, we should be transitioned by then.

2. I don't think the caching is worth it. It uses up memory for no real benefit that I can see (it's just caching the result of a simple array transformation, not something expensive like a DB query). I predict this function will be less and less used in the future as unique names get phased out.

Reply:
OK - I guess I can accept that....

Reply:
This would be hit by every test under the sun....